hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerElementCollectionForEmbeddableEntityTypeListTest.java

```diff
@@ -0,0 +1,815 @@
+
+		test( context,
+			  session.find( Person.class, thePerson.getId() )
+					  .invoke( found -> {
+						  context.assertFalse( found.getPhones().isEmpty() );
+						  found.setPhones( null );
+					  } )
+					  .call( session::flush )
+					  .chain( () -> openMutinySession().find( Person.class, thePerson.getId() ) )
+					  .invoke( foundPerson -> assertPhones( context, foundPerson ) )
+		);
+	}
+
+	/**
+	 * With the Stage API, run a native query to check the content of the table containing the collection of elements
+	 * associated to the selected person.
+	 */
+	private CompletionStage<List<Object>> selectFromPhonesWithStage(Person person) {
+		return openSession()
+				.createNativeQuery( "SELECT * FROM Person_phones where Person_id = ?" )
+				.setParameter( 1, person.getId() )
+				.getResultList();
+	}
+
+	/**
+	 * With the Mutiny API, run a native query to check the content of the table containing the collection of elements
+	 * associated to the selected person
+	 */
+	private Uni<List<Object>> selectFromPhonesWithMutiny(Person person) {
+		return openMutinySession()
+				.createNativeQuery( "SELECT * FROM Person_phones where Person_id = ?" )
+				.setParameter( 1, person.getId() )
+				.getResultList();
+	}
+
+	private static void assertPhones(TestContext context, Person person, String... phones) {
+		context.assertNotNull( person );
+		context.assertEquals( phones.length, person.getPhones().size() );
+		for ( String number : phones) {
+			context.assertTrue( phonesContainNumber( person, number) );
+		}
+	}
+
+	private static boolean phonesContainNumber(Person person, String phoneNum) {
+		for ( Phone phone : person.getPhones() ) {
+			if ( phone.getNumber().equals( phoneNum ) ) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Entity(name = "Person")
+	public static class Person {
+
+		@Id
+		private Integer id;
+
+		private String name;
+
+		@ElementCollection(fetch = FetchType.EAGER)
+		private List<Phone> phones = new ArrayList<>();
+
+		public Person() {
+		}
+
+		public Person(Integer id, String name, List<Phone> phones) {
+			this.id = id;
+			this.name = name;
+			this.phones = phones;
+		}
+
+		public Integer getId() {
+			return id;
+		}
+
+		public void setId(Integer id) {
+			this.id = id;
+		}
+
+		public String getName() {
+			return name;
+		}
+
+		public void setName(String name) {
+			this.name = name;
+		}
+
+		public Phone getPhone(String number) {
+			for( Phone phone : getPhones() ) {
+				if( phone.getNumber().equals( number ) ) {
+					return phone;
+				}
+			}
+			return null;
+		}
+
+		public void setPhones(List<Phone> phones) {
+			this.phones  = phones;
+		}
+
+		public List<Phone> getPhones() {
+			return phones;
+		}
+	}
+
+	@Embeddable
+	public static class Phone {
+
+		private String number;
+
+		private String country;
+
+		// FIXME: add nested @Embeddable
+		// @ElementCollection(fetch = FetchType.EAGER)
+		// private List<AlternativePhone> alternatives = new ArrayList<>();
```

Comment:
Can you remove this comments from the version we are going to merge?

Reply:
This is probably going to be a different test class anyway

Reply:
yes.. removed.  Already started on setting up the different test class.